extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,261 @@
+use `passphrase` as a default credential type.
+
+The IRC server MAY forward the `REG CREATE` command to a central authority, or process it locally. Clients
+SHOULD NOT be sent unnecessary legacy (e.g. from NickServ) private messages or notices in response to `REG`
+commands, where native numerics such as the ones defined in this document can replace them.
+
+Upon success, the IRC server MUST send the `RPL_REGISTRATION_SUCCESS` numeric, which looks like:
+
+| No. | Label                      | Format                                                         |
+| --- | -------------------------- | -------------------------------------------------------------- |
+| 920 | `RPL_REGISTRATION_SUCCESS` | `:<server> 920 <user_nickname> <accountname> :Account created` |
+
+If the server requires a verification token, it MUST also reply with the `RPL_REG_VERIFICATION_REQUIRED`
+numeric, which looks like:
+
+| No. | Label                           | Format                                                                                                       |
+| --- | ------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| 927 | `RPL_REG_VERIFICATION_REQUIRED` | `:<server> 927 <user_nickname> <accountname> <callback_namespace:><callback> :A verification token was sent` |
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these numerics:
+
+| No. | Label                        | Format                                                                                          |
+| --- | ---------------------------- | ----------------------------------------------------------------------------------------------- |
+| 921 | `ERR_ACCOUNT_ALREADY_EXISTS` | `:<server> 921 <user_nickname> <accountname> :Account already exists`                           |
+| 922 | `ERR_REG_UNSPECIFIED_ERROR`  | `:<server> 922 <user_nickname> <accountname> :<descriptive_text>`                               |
+| 929 | `ERR_REG_INVALID_CALLBACK`   | `:<server> 929 <user_nickname> <accountname> <callback> :Callback token is invalid`             |
+| 928 | `ERR_REG_INVALID_CRED_TYPE`  | `:<server> 928 <user_nickname> <accountname> <cred_type> :Credential type is invalid`           |
+| 440 | `ERR_REG_UNAVAILABLE`        | `:<server> 440 <user_nickname> <subcommand> :Authentication services are currently unavailable` |
+
+The server MAY send additional informative text upon registration success or failure using `RPL_TEXT` or `NOTICE`.
+
+## The `REG VERIFY` sub-command
+
+The `REG VERIFY` command signals the intent of a client to submit a verification token for their
+account registration.  If a valid callback is specified by the client during the `REG CREATE`
+command, the server MUST send the token to the specified callback.
+
+A `REG VERIFY` command consists of the following format:
+
+    REG VERIFY <accountname> <auth_code>
```

Comment:
I think this should include the (optional?) password if we're going to immediately log the user in [L111]. Or indicate that this command is only available while already logged in (how do you log in?) (in which case the sentence below "the irc server should also send an rpl_loggedin ..." would no longer be valid)

There are many users who accidentally paste their authentication code in chat by accident.